src/connect/ConnectWizard.ts

```diff
@@ -202,8 +204,36 @@ export class ConnectWizard {
             activeItem: serviceChoices[0]
         });
         this._result.resourceName = pick.label;
+        // get the list of containers
```

Comment:
💡 Can we simplify all this with:

I would say think this through carefully, but avoid umbrella else and probably fail early, below if just sample, also avoids the nested `if`

```*.ts
       // get the list of containers
        const podName = await kubectlClient.getPodName(this._result.resourceName);
        const containersList: string[] = await kubectlClient.getContainersList(podName);

        if (!containersList || !containersList.length) {
            // containerList or containerlist.length are falsy
            // ⇒ do not attempt to process containerlist
            return (input: MultiStepInput) => this.inputPortsAsync(input, resourceType);
        }
        
        if (containersList.length > 1) {
           // show containers quick pick list
           const containerChoices: vscode.QuickPickItem[] = containersList.map(containers => ({ label: containers }));
           return (input: MultiStepInput) => this.inputContainersAsync(input, containerChoices, resourceType);
        } else if (containersList.length === 0) {
            // single container for the pod selected
            this._result.containerName = containersList[0];
            return (input: MultiStepInput) => this.inputPortsAsync(input, resourceType);
        }
```

Reply:
updated, this.